internal/dag/cache.go

```diff
@@ -425,3 +425,36 @@ func (kc *KubernetesCache) LookupSecret(name types.NamespacedName, validate func
 
 	return s, nil
 }
+
+func (kc *KubernetesCache) DelegationPermitted(secret types.NamespacedName, to string) bool {
```

Comment:
Nit: Perhaps `secretName` and `targetNamespace` would be clearer variable names. `secret` works ok, but it took me a few minutes to understand what `to` meant.

Reply:
ok cool I renamed. These were just lifted and shifted from the existing code, so wasn't changing much, but internal names won't be a problem.